controllers/buildpipeline/buildpipeline_controller.go

```diff
@@ -104,7 +106,23 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	if err != nil {
 		logger.Error(err, "Failed to get Application from Component",
 			"Component.Name ", component.Name, "Component.Namespace ", component.Namespace)
-		return ctrl.Result{}, err
+		msg := fmt.Sprint("Failed to get Application from Component",
+			"Component.Name ", component.Name, "Component.Namespace ", component.Namespace,
+			"Error:", err)
+		patch := client.MergeFrom(pipelineRun.DeepCopy())
+		jsonResult, jserr := json.Marshal(map[string]string{
+			"status":  "failure",
+			"message": msg,
+		})
+		if jserr != nil {
+			return ctrl.Result{}, jserr
+		}
+
+		_ = metadata.SetAnnotation(&pipelineRun.ObjectMeta, helpers.CreateSnapshotAnnotationName, string(jsonResult))
+		err = r.Client.Patch(ctx, pipelineRun, patch)
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
```

Comment:
Can we not duplicate the whole function here please? I belive we could factro out some code form adapter and make it usable on both places.

Reply:
I would love to make those functions into single one, I'm just running into the case with different recievers (Adapter vs Reconciler) and I haven't found solution for function with multiple of different recievers for one definition. If you know about a solution for this one, I'm all in for that.
Otherwise it looks like we will end up with simillar functions.

Reply:
ofc, you must pass client and context as parameter